app/models/order.rb

```diff
@@ -32,4 +33,30 @@ def total_items
   def set_total
     update(total: line_items.to_a.sum(&:price))
   end
+
+  def received
+    update(status: 'received', placed_on: Time.now)
+  end
+
+  def check_branch_inventory?
+    meal_max_quantity = meals.joins(ingredients_meals: :inventories)
+          .where(inventories: { branch_id: branch_id })
+          .group(:id).minimum('inventories.quantity/ingredients_meals.ingredient_quantity')
+    line_items.joins(:meal).each do |line_item|
+      if (line_item.quantity > meal_max_quantity[line_item.meal_id]) && meal_max_quantity[line_item.meal_id] == 0
+        errors.add(:base, "#{line_item.meal.name} is out of stock")
+      elsif line_item.quantity > meal_max_quantity[line_item.meal_id]
+        errors.add(:base, "Only #{meal_max_quantity[line_item.meal.id]} quantity of #{line_item.meal.name} can be added")
+      end
+    end
+    errors.empty?
+  end
+
+  def update_inventory(inc)
```

Comment:
inc parameter should be named parameter. It's hard to track when you see true/false in calling arguments

Reply:
Done